kinto/core/storage/memory.py

```diff
@@ -322,7 +322,7 @@ def extract_record_set(records, filters, sorting,
         paginated = {}
         for rule in pagination_rules:
             values = apply_filters(filtered, rule)
-            paginated.update(((x[id_field], x) for x in values))
+            paginated.update((id(x), x) for x in values)
```

Comment:
index by memory address? I realize I don't understand why we don't just build a list :)

Reply:
Well, I just tried building a list and none of the tests failed. Maybe the person who wrote it originally was afraid that different pagination rules would re-match a record twice? In that case, the tests probably won't fail because we don't construct pagination rules in such a way where this could happen (but a pathological user could conceptually find a way if they were motivated). Since none of the tests fail, I'll take it out..